bottlecap/src/traces/propagation/carrier.rs

```diff
@@ -0,0 +1,68 @@
+/// Code inspired, and copied, by OpenTelemetry Rust project.
+/// <https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/propagation/mod.rs>
+///
+use std::collections::HashMap;
+
+/// Injector provides an interface for a carrier to be used
+/// with a Propagator to inject a Context into the carrier.
+///
+pub trait Injector {
```

Comment:
Same for these 2. Can we avoid/postpone creating extra traits?

Reply:
This allows us to directly implement this traits for carriers which we might not have control of, for example, here I tend to convert `hyper::HeaderMap` into a `HashMap` so I can use the `extract` and `inject` methods on the struct right away.

At the top of the file I explain that I directly copied this from the OTel rust implementation for extractor/injectors.

I think this would help us whenever we have to expand, and its also very generic, what do you think of this?